python/lsst/ts/observatory/control/maintel/mtcs.py

```diff
@@ -579,6 +579,37 @@ async def dome_el_in_position(self) -> str:
         await self._dome_el_in_position.wait()
         return "Dome elevation in position."
 
+    async def wait_for_dome_ready(self) -> None:
+        """Wait for the dome to be in a state where it can be moved."""
+        dome_state = await self.rem.mtdome.evt_summaryState.aget(
+            timeout=self.fast_timeout
+        )
+        if dome_state.summaryState != salobj.State.ENABLED:
+            raise RuntimeError("Dome is not enabled. Cannot move.")
+
+        self.log.info("Dome is ready for movement.")
+
+    async def park_dome(self) -> None:
+        """Park the dome by moving it to the park azimuth."""
+        self.log.info(f"Parking dome at azimuth {self.dome_park_az}.")
+
+        # Ensure the dome is ready for a move
+        await self.wait_for_dome_ready()
+
+        # Move the dome to the park position
+        await self.rem.mtdome.cmd_park.set_start(timeout=self.long_timeout)
+
+        # Wait for the dome to emit the evt_azMotion event in the PARKED state
+        self.log.info("Waiting for dome to reach the PARKED status.")
+
+        az_motion = await self.rem.mtdome.evt_azMotion.next()
```

Comment:
you will need to make this check in a loop because there may be intermediary states between the current state and PARKED. You will need to define "valid states" (which are ok to transition to while parking) and "bad states" (which are not ok to transition to while parking and should raise an exception).  

See, for instance, how this is done in [_wait_for_mtm1m3_detailed_state](https://github.com/lsst-ts/ts_observatory_control/blob/9e9ab43b4474a69c4e12723776bfbb1c55e9a023/python/lsst/ts/observatory/control/maintel/mtcs.py#L871).

Reply:
7390752ef6738cd30f1ee84c9e65ffc1a28a7aaf